### PR TITLE
Update plotting.py

### PR DESCRIPTION
**Pull request by cjhopp**
Rework detection_multiplot().

### CODE REVIEW

**Comment by cjhopp** [comment]
Hold off on this for a second or two...

**Comment by calum-chamberlain** [comment]
Ace, so the process will be, make sure the tests all pass (they should), then I will give it a quick review and we can then merge. If you want to make any changes you can continue to commit to this branch and the pull request will update.

**Comment by calum-chamberlain** [comment]
Also, could you make a note in the CHANGES.md file outlining (briefly) what you did so we can keep a track of the bug-fixes :)

**Comment by cjhopp** [comment]
Aight. Should be good to go. I was getting intermittent errors, but was due to bad input data on my end.

**Comment by calum-chamberlain** [comment]
Looks like a pep8 fail.

**Comment by cjhopp** [comment]
Issue with the lambda...? And what's the issue with the last formatting string..?

**Comment by calum-chamberlain** [comment]
Give us some links boyo.

**Comment by cjhopp** [comment]
Lambda issue here: https://github.com/eqcorrscan/EQcorrscan/blob/fbd2ebd3281e9fa9e0fb90b32aab941f559aa5b3/eqcorrscan/utils/plotting.py#L823-L825

**Comment by cjhopp** [comment]
String formatting here: [https://github.com/eqcorrscan/EQcorrscan/blob/fbd2ebd3281e9fa9e0fb90b32aab941f559aa5b3/eqcorrscan/utils/plotting.py#L883-L887](url)

**Comment by cjhopp** [comment]
Must be better way to share links...

**Comment by calum-chamberlain** [comment]
Link to the failed test! That looks like it will have failed because the line is too long... Does the test that failed say 'line too long'....
And there is a better way: [pep8 fail](https://ci.appveyor.com/project/calum-chamberlain/eqcorrscan-jsycv/build/1.0.180)

**Comment by cjhopp** [comment]
Isn't that link available via 'Details' next to the pertinent test?

**Comment by calum-chamberlain** [comment]
It is indeed, but the one that I looked at was an obvious pep8 fail - I thought you had something different? There are 12 test sessions run and I am too lazy to look at all of them.

**Comment by cjhopp** [comment]
Let's try that again...

**Comment by calum-chamberlain** [comment]
Looks like everything should pass - I think the codecov/patch might fail, but that's because we don't report coverage on plotting.  I will give this a quick review tomorrow. :+1:

File: eqcorrscan/utils/plotting.py [hunk 1]
```diff
@@ -852,43 +853,42 @@ def detection_multiplot(stream, template, times, streamcolour='k',
...
-                      template_tr.data * normalizer,
-                      templatecolour, linewidth=1.2)
+        time = times[i]
+        time_dt = time.datetime
+        template_times = [time_dt +
+                          dt.timedelta((j * template_tr.stats.delta) /
+                                       86400)
+                          for j in range(len(template_tr.data))]
+        # Normalize the template according to the data detected in
+        normalizer = max(image.data[int((template_times[0] -
```
Comment (calum-chamberlain) [inline eqcorrscan/utils/plotting.py]: Pretty sure there is a prettier way to write this, but this is fine.

File: eqcorrscan/utils/plotting.py [hunk 2]
```diff
@@ -852,43 +853,42 @@ def detection_multiplot(stream, template, times, streamcolour='k',
...
+                                        image_times[0]).
+                                        total_seconds() /
+                                        image.stats.delta):
+                                    int((template_times[-1] -
+                                         image_times[0]).
+                                        total_seconds() /
+                                        image.stats.delta)] /
+                         max(image.data))
+        normalizer /= max(template_tr.data)
```
Comment (calum-chamberlain) [inline eqcorrscan/utils/plotting.py]: You removed the try/except statement, any reason for this? I think it looks (from my comment) like it was necessary when there was no data present?

File: eqcorrscan/utils/plotting.py [hunk 3]
```diff
@@ -852,43 +853,42 @@ def detection_multiplot(stream, template, times, streamcolour='k',
...
+                  templatecolour, linewidth=1.2)
         ylab = '.'.join([template_tr.stats.station,
                          template_tr.stats.channel])
         axis.set_ylabel(ylab, rotation=0,
                         horizontalalignment='right')
-        # axis.yaxis.set_ticks([])
+        axis.yaxis.set_ticks([])
     if len(template) > 1:
-        axes[len(axes) - 1].set_xlabel('Time')
+        axes[-1].set_xlabel('Time (s)')
```
Comment (calum-chamberlain) [inline eqcorrscan/utils/plotting.py]: Based on the mdates formatter below, s isn't the correct unit.

File: eqcorrscan/utils/plotting.py [hunk 4]
```diff
@@ -852,43 +853,42 @@ def detection_multiplot(stream, template, times, streamcolour='k',
...
+        axes[-1].xaxis.set_major_locator(mdates.SecondLocator())
+        axes[-1].xaxis.set_major_formatter(mdates.DateFormatter('%H:%M:%S'))
+        xaxis.set_major_formatter(mdates.DateFormatter('%H:%M:%S'))
     else:
         axis.set_xlabel('Time')
+        axis.xaxis.set_major_locator(mdates.SecondLocator())
+        axis.xaxis.set_major_formatter(mdates.DateFormatter('%H:%M:%S'))
     plt.subplots_adjust(hspace=0, left=0.175, right=0.95, bottom=0.07)
-    plt.xticks(rotation=10)
+    # plt.xticks(rotation=10)
```
Comment (calum-chamberlain) [inline eqcorrscan/utils/plotting.py]: I had the rotation there so plot-labels didn't overlap, is that not an issue now?

File: eqcorrscan/utils/plotting.py (hunk 1 above)
Comment (cjhopp) [inline eqcorrscan/utils/plotting.py]: I didn't mess with any of the normalizing. But I think it's fine.

File: eqcorrscan/utils/plotting.py (hunk 4 above)
Comment (cjhopp) [inline eqcorrscan/utils/plotting.py]: So for plots with less than ~10 seconds the way it's written currently works well. Maybe it's too much to assume that templates will always be of a certain length (<3 sec or so), and will therefore be best plotted on axes of less than 10 seconds? The only reason it even came to mind is that Martha complained about the diagonal labels on a cumulative detection plot once.

**Comment by cjhopp** [comment]
......I'm an idiot. In looking at the example .png you already have, I realized that the entire purpose is to allow multiple detections to be plotted on one trace.

So the way it was written was mostly fine, with maybe a minor tweak to the documentation and maybe in what the argument 'times' consists of...

**Pull request closed without merging**